packages/internal-test-utils/consoleMock.js

```diff
@@ -382,8 +382,9 @@ export function createLogAssertion(
 
         // Main logic to check if log is expected, with the component stack.
         if (
-          normalizedMessage === expectedMessage ||
-          normalizedMessage.includes(expectedMessage)
+          typeof expectedMessage === 'string' &&
+          (normalizedMessage === expectedMessage ||
```

Comment:
This might forbid `console.error('undefined')` and `assertConsoleErrorDev([undefined])` but that doesn't seem like a valid scenario anyway.